network.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 go-dockerclient authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package docker
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// Network represents a network.
+type Network struct {
+	Name      string      `json:"name"`
+	ID        string      `json:"id"`
+	Type      string      `json:"type"`
+	Endpoints []*Endpoint `json:"endpoints"`
+}
+
+// Endpoint represents an endpoint.
+type Endpoint struct {
+	Name    string `json:"name"`
+	ID      string `json:"id"`
+	Network string `json:"network"`
+}
+
+// Service represents a service.
+type Service struct {
+	ID string `json:"id"`
+}
+
+// ListNetworks returns all networks.
+func (c *Client) ListNetworks() ([]Network, error) {
+	path := "/networks"
+	body, _, err := c.do("GET", path, doOptions{})
+	if err != nil {
+		return nil, err
+	}
+	var networks []Network
+	err = json.Unmarshal(body, &networks)
+	if err != nil {
+		return nil, err
+	}
+	return networks, nil
+}
+
+// NetworkInfo returns information about a network by its ID.
+func (c *Client) NetworkInfo(id string) (*Network, error) {
+	path := "/networks/" + id
+	body, status, err := c.do("GET", path, doOptions{})
+	if status == http.StatusNotFound {
+		return nil, &NoSuchNetwork{ID: path}
+	}
+	if err != nil {
+		return nil, err
+	}
+	var network Network
+	err = json.Unmarshal(body, &network)
+	if err != nil {
+		return nil, err
+	}
+	return &network, nil
+}
+
+// NoSuchNetwork is the error returned when a given network does not exist.
+type NoSuchNetwork struct {
+	ID  string
+	Err error
```

Comment:
I mainly copied NoSuchContainer, which also has Err and doesn't appear to use it.  I'm happy to take it out of NoSuchNetwork, but should I be consistent with NoSuchContainer, or should I remove the unused bit?